libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts

```diff
@@ -20,35 +30,82 @@ import { coerceElement } from '@angular/cdk/coercion';
     standalone: true,
     imports: [PortalModule],
-    template: ` <ng-template [cdkPortalOutlet]="portal"></ng-template>`
+    template: ` <ng-template cdkPortalOutlet></ng-template>`,
+    providers: [DestroyedService]
 })
-export class DynamicPortalComponent implements OnChanges {
+export class DynamicPortalComponent implements AfterViewInit {
     /** The DOM element to attach */
-    @Input() domElement: HTMLElement | ElementRef<HTMLElement>;
+    @Input()
+    set domElement(value: HTMLElement | ElementRef<HTMLElement> | string) {
+        this.portalContent$.next(value);
+    }
 
     /** The component to attach */
     @Input()
-    component: ComponentType<any>;
+    set component(value: ComponentType<any>) {
+        this.portalContent$.next(value);
+    }
 
     /** The template to attach */
     @Input()
-    template: TemplateRef<any>;
+    set template(value: TemplateRef<any>) {
+        this.portalContent$.next(value);
+    }
 
-    /** Portal associated with the Portal outlet. */
-    portal?: Portal<any>;
+    /** The Content which should be attached and can be any of the items */
+    @Input()
+    set auto(value: DynamicPortalComponent['domElement' | 'component' | 'template']) {
+        this.portalContent$.next(value);
+    }
+
+    /** @hidden */
+    @ViewChild(CdkPortalOutlet)
+    portalOutlet?: CdkPortalOutlet;
+
+    /** @hidden */
+    private portalContent$ = new BehaviorSubject<
+        DynamicPortalComponent['domElement' | 'component' | 'template'] | undefined
+    >(undefined);
+
+    /** @hidden */
+    private destroy$ = inject(DestroyedService);
 
     /** @hidden */
     private viewContainerRef = inject(ViewContainerRef);
 
     /** @hidden */
-    ngOnChanges(changes: SimpleChanges): void {
-        this.portal?.detach();
-        if (changes['domElement']) {
-            this.portal = new DomPortal(coerceElement(this.domElement));
-        } else if (changes['component']) {
-            this.portal = new ComponentPortal(this.component);
-        } else if (changes['template']) {
-            this.portal = new TemplatePortal(this.template, this.viewContainerRef);
-        }
+    private renderer = inject(Renderer2);
+
+    /** @hidden */
+    private elementRef = inject(ElementRef);
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        const portalOutlet = this.portalOutlet as CdkPortalOutlet;
+        this.portalContent$
+            .pipe(
+                map((content) => {
+                    portalOutlet.detach();
+                    if (content) {
+                        if (typeof content === 'string') {
+                            const fdTextElement = this.renderer.createElement('p');
+                            fdTextElement.classList.add('fd-text');
+                            fdTextElement.innerHTML = content;
+                            this.renderer.appendChild(this.elementRef.nativeElement, fdTextElement);
+                            return new DomPortal(fdTextElement);
```

Comment:
Not sure if this should be here. Logic looks out of scope of this component

Reply:
Yes, you are totally right